File: net/net-registri-log/ApiLog/Models/ApiLog.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace net_registri_log.ApiLog.Models
{
    public class ApiObject
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public string UserId { get; set; }
        [MaxLength(8)]
        public string Method { get; set; }
        [MaxLength(2083)]
        public string Url { get; set; }
        [MaxLength(1024)]
        public string QueryString { get; set; }
        public string RequestBody { get; set; }
        /// <summary>
        /// Body size in kb.
        /// </summary>
        public float? RequestSize { get; set; }
        public string ResponseBody { get; set; }
        /// <summary>
        /// Body size in kb.
        /// </summary>
        public float? ResponseSize { get; set; }
        public long ElapsedMilliseconds { get; set; }
    }
}

File: node/projects/ngx-registri-log/src/lib/api-log/api-log.ts
export class ApiLog {
    id: number;
    date: Date;
    userId: string;
    method: string;
    url: string;
    queryString: string;

    requestBody: string;
    requestBodyObject: object;
    /// <summary>
    /// Body size in kb.
    /// </summary>
    requestSize: number;
    
    responseBody: string;
    responseBodyObject: object;
    /// <summary>
    /// Body size in kb.
    /// </summary>
    responseSize: number;
    
    elapsedMilliseconds: number;

    constructor(data?: Partial<ApiLog>) {
        Object.assign(this, data);
        this.id = (+this.id || 0);
        if (this.requestSize != null)
            this.requestSize = +this.requestSize.toFixed(1);
        
            if (this.responseSize != null)
            this.responseSize = +this.responseSize.toFixed(1);
        
        if (this.requestBody)
            this.requestBodyObject = JSON.parse(this.requestBody);
        
        if (this.responseBody)
            this.responseBodyObject = JSON.parse(this.responseBody);
    }
}